app/components/SystemData.jsx

```diff
@@ -1,36 +1,66 @@
 import React, { Component } from 'react';
 import StopLight from './metrics/StopLight';
 import Graph from './metrics/Graph';
+import { time } from 'console';
 
 class SystemData extends Component {
   // state should probably go in this component
   render() {
-    // const avg =
-    //   Math.round(
-    //     (this.props.data.sum / this.props.data.numOfDataPoints) * 100
-    //   ) / 100; // Round to 2 decimals
-    const { title } = this.props;
-    const { sum, numOfDataPoints, smallest, largest } = this.props.data;
-    const avg =
-      sum === 0 && numOfDataPoints === 0
-        ? 'None'
-        : (sum / numOfDataPoints).toFixed(2);
+             // const avg =
+             //   Math.round(
+             //     (this.props.data.sum / this.props.data.numOfDataPoints) * 100
+             //   ) / 100; // Round to 2 decimals
+             const { title } = this.props;
+             // if title === 'data size' do the following:
+             if (title === 'Data Size' || title === 'Processing Time') {
```

Comment:
I think this feature can be implemented without this messy conditional logic. What if we track pendingDuration in the same way we track the other metrics? That way we can display average, smallest, and largest. This also allows us to keep our code DRY.

Reply:
ok- we tried this and were successful, however, the algorithm is not working correctly for "pending duration"